controllers/reconcilers/kisvc_route_reconciler.go

```diff
@@ -0,0 +1,206 @@
+		routeHandler:   components.NewRouteHandler(client, ctx, logger),
+		deltaProcessor: processors.NewDeltaProcessor(),
+	}
+}
+
+func (r *kserveInferenceServiceRouteReconciler) Reconcile() error {
+
+	// Create Desired resource
+	desiredResource, err := r.createDesiredResource()
+	if err != nil {
+		return err
+	}
+
+	// Get Existing resource
+	existingResource, err := r.getExistingResource()
+	if err != nil {
+		return err
+	}
+
+	// Process Delta
+	if err = r.processDelta(desiredResource, existingResource); err != nil {
+		return err
+	}
+	return nil
+}
+
+func (r *kserveInferenceServiceRouteReconciler) createDesiredResource() (*v1.Route, error) {
+	ingressConfig, err := kservev1beta1.NewIngressConfig(r.client)
+	if err != nil {
+		return nil, err
+	}
+
+	disableIstioVirtualHost := ingressConfig.DisableIstioVirtualHost
+	if disableIstioVirtualHost == false {
+
+		serviceHost := getServiceHost(r.isvc)
+		isInternal := false
+		//if service is labelled with cluster local or knative domain is configured as internal
+		if val, ok := r.isvc.Labels[constants.VisibilityLabel]; ok && val == constants.ClusterLocalVisibility {
+			isInternal = true
+		}
+		serviceInternalHostName := network.GetServiceHostname(r.isvc.Name, r.isvc.Namespace)
+		if serviceHost == serviceInternalHostName {
+			isInternal = true
+		}
+		if isInternal {
+			return nil, nil
+		}
+
+		if ingressConfig.PathTemplate != "" {
+			serviceHost = ingressConfig.IngressDomain
+		}
+		annotations := utils.Filter(r.isvc.Annotations, func(key string) bool {
+			return !utils.Includes(constants.ServiceAnnotationDisallowedList, key)
+		})
+		route := &v1.Route{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:        r.isvc.Name,
+				Namespace:   IstioNamespace,
+				Annotations: annotations,
+				Labels:      r.isvc.Labels,
+			},
+			Spec: v1.RouteSpec{
+				Host: serviceHost,
+				To: v1.RouteTargetReference{
+					Kind:   "Service",
+					Name:   IstioIngressService,
+					Weight: pointer.Int32(100),
+				},
+				Port: &v1.RoutePort{
+					TargetPort: intstr.FromString("https"),
+				},
+				TLS: &v1.TLSConfig{
+					Termination:                   v1.TLSTerminationPassthrough,
+					InsecureEdgeTerminationPolicy: v1.InsecureEdgeTerminationPolicyRedirect,
+				},
+				WildcardPolicy: v1.WildcardPolicyNone,
+			},
+			Status: v1.RouteStatus{
+				Ingress: []v1.RouteIngress{},
+			},
+		}
+		return route, nil
+	}
+	return nil, nil
+}
+
+func (r *kserveInferenceServiceRouteReconciler) getExistingResource() (*v1.Route, error) {
+	return r.routeHandler.FetchRoute(types.NamespacedName{Name: r.isvc.Name, Namespace: IstioNamespace})
+}
+
+func (r *kserveInferenceServiceRouteReconciler) processDelta(desiredRoute *v1.Route, existingRoute *v1.Route) (err error) {
+	comparator := r.routeHandler.GetComparator()
+	delta := r.deltaProcessor.ComputeDelta(comparator, desiredRoute, existingRoute)
+
+	if !delta.HasChanges() {
+		r.log.Info("No delta found")
+		return nil
+	}
+
+	if delta.IsAdded() {
```

Comment:
Please, properly set owner references.

Reply:
In this case route could not have owner reference because its created in `istio-system` namespace. Owner references could only be assigned within same namespaces.